backend/baseclass.pm

```diff
@@ -855,7 +855,11 @@ sub proxy_console_call {
         # Move the decision to actually die to the server side instead.
         # For this ignore backend::baseclass::die_handler.
         local $SIG{__DIE__} = 'DEFAULT';
-        $wrapped_result->{result} = $console->$function(@$args);
+        if ($wrapped_call->{wantarray}) {
+            $wrapped_result->{result} = [$console->$function(@$args)];
+        } else {
+            $wrapped_result->{result} = $console->$function(@$args);
+        }
```

Comment:
Can you make that a single assignment call with the ternary operator?

```suggestion
        $wrapped_result->{result} = $wrapped_call->{wantarray} ? [$console->$function(@$args)] : $console->$function(@$args);
```

Reply:
sure, I took the long version, because the line get so long and thought it is more readable. But I don't have hard feelings.